File: backend/app/schemas/insight.py
from datetime import date

from pydantic import BaseModel


class CategoryAverage(BaseModel):
    category: str
    average_severity: float


class InsightsSummaryResponse(BaseModel):
    total_entries: int
    days_tracked: int
    average_severity_per_category: list[CategoryAverage]
    most_frequent_category: str | None
    highest_avg_severity_category: str | None
    entries_last_7_days: int


class TimelinePoint(BaseModel):
    date: date
    average_severity: float


class TimelineResponse(BaseModel):
    category: str
    points: list[TimelinePoint]

File: frontend/src/types/insight.ts
export type CategoryAverage = {
  category: string;
  average_severity: number;
};

export type InsightsSummary = {
  total_entries: number;
  days_tracked: number;
  average_severity_per_category: CategoryAverage[];
};

export type TimelinePoint = {
  date: string;
  average_severity: number;
};

export type TimelineResponse = {
  category: string;
  points: TimelinePoint[];
};